### DISCUSSION TITLE
Login with IDP available on user account (not domain)

### DISCUSSION THREAD

**Discussion by kadhemmejri** [keycloak/keycloak#8747] [Q&A]
Hello,

Please I need your help

Is there any possibility to configure an authentication flow like this :

The first login page asks for username/email and when the user click next he gets to choose with which identity provider he wants to connect from the list of identity providers available on the user account (Identity provider link) 

Scenario :

Imagine the user has 2 IDP configured on his profile (the domain can have more IDP configured)

![IDP list](https://user-images.githubusercontent.com/70603515/140926861-20442eef-8ae7-4737-9086-4b7fd85ac1e8.PNG)

When the user open the login page, he introduce the login/email and click next to arrive on an interface with "choose an identity provider" with a list of identity providers added to the user account 
![login page](https://user-images.githubusercontent.com/70603515/140926859-b707b75e-9fb6-4b84-93bf-26da4aab29ad.png)
Could you tell me if this can be configured with any version of Keycloak ? or should we make specific developpement 

Thank you so much for your prompt reply

**Comment by kadhemmejri** [comment]
Hello,

any idea please ?

**Comment by kadhemmejri** [comment]
well...

**Comment by jbman** [comment]
This is slightly related to idea https://github.com/keycloak/keycloak/discussions/8839 which also requires a screen that requests the users username or email.
Maybe you could implement it with a custom [Authenticator](https://github.com/keycloak/keycloak/blob/main/server-spi-private/src/main/java/org/keycloak/authentication/Authenticator.java) which is then a added in the Browser authentication flow (see menu item Authentication, Tab 'Flow' and select flow 'Browser').
